router/h2/src/main/scala/io/buoyant/router/h2/StreamStatsFilter.scala

```diff
@@ -41,46 +44,118 @@ class StreamStatsFilter(statsReceiver: StatsReceiver)
         case Throw(_) => failures.incr()
       }
     }
+
   }
 
-  //   total number of requests received
+  class FrameStats(
```

Comment:
I think we could simplify by moving the contents of StreamStats into this class, deleting StreamStats, and renaming this to StreamStats.